lib/src/screens/notifications/notifications_screen.dart

```diff
@@ -36,51 +34,63 @@ class _NotificationsScreenState extends State<NotificationsScreen> {
     final lang = FlutterI18n.currentLocale(context)!.languageCode;
 
-    void _onNotificationTap(Notification notification) {
-      final action = notification.action;
-
-      if (action != null && action is CredentialDetailNavigationAction) {
-        final repo = IrmaRepositoryProvider.of(context);
-        final credType = repo.irmaConfiguration.credentialTypes[action.credentialTypeId]!;
-        final translatedAttributeType = credType.name.translate(lang);
+    if (action != null && action is CredentialDetailNavigationAction) {
+      final repo = IrmaRepositoryProvider.of(context);
+      final credType = repo.irmaConfiguration.credentialTypes[action.credentialTypeId]!;
+      final translatedAttributeType = credType.name.translate(lang);
 
-        _notificationsBloc.add(
-          MarkNotificationAsRead(notification.id),
-        );
+      _notificationsBloc.add(
+        MarkNotificationAsRead(notification.id),
+      );
 
-        Navigator.of(context).push(
-          MaterialPageRoute(
-            builder: (context) => CredentialsDetailScreen(
-              categoryName: translatedAttributeType,
-              credentialTypeId: action.credentialTypeId,
-            ),
+      Navigator.of(context).push(
+        MaterialPageRoute(
+          builder: (context) => CredentialsDetailScreen(
+            categoryName: translatedAttributeType,
+            credentialTypeId: action.credentialTypeId,
           ),
-        );
-      }
+        ),
+      );
     }
+  }
 
-    void _onNotificationDismiss(Notification notification) => _notificationsBloc.add(
-          SoftDeleteNotification(notification.id),
-        );
-
-    Widget _emptyListIndicator() => Padding(
-          padding: EdgeInsets.all(theme.defaultSpacing),
-          child: Row(
-            mainAxisAlignment: MainAxisAlignment.center,
-            children: const [
-              Flexible(
-                  child: TranslatedText(
-                'notifications.empty',
-                textAlign: TextAlign.center,
-              ))
-            ],
-          ),
-        );
+  void _onNotificationDismiss(Notification notification) => _notificationsBloc.add(
+        SoftDeleteNotification(notification.id),
+      );
+
+  Widget _emptyListIndicator(IrmaThemeData theme) =>
+      // It needs to be wrapped in a ListView because of the RefreshIndicator
+      ListView(
+        children: [
+          Padding(
+            padding: EdgeInsets.all(theme.defaultSpacing),
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: const [
+                Flexible(
+                    child: TranslatedText(
+                  'notifications.empty',
+                  textAlign: TextAlign.center,
+                ))
+              ],
+            ),
+          )
+        ],
+      );
+
+  Future<void> _onRefresh() async {
+    _notificationsBloc.add(
+      LoadNotifications(),
+    );
+  }
```

Comment:
Small detail, but if you want this function to be async then you probably forgot the `await`. If nothing has to actually wait on this operation, then you can also consider fully removing the async keyword.